skimage/draw/draw_nd.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+
+
+def _round_safe(coords):
+    """Round coords while ensuring successive values are less than 1 apart.
+
+    When rounding coordinates for `line_nd`, we want coordinates that are less
+    than 1 apart (always the case, by design) to remain less than one apart.
+    However, NumPy rounds values to the nearest *even* integer, so:
+
+    >>> np.round([0.5, 1.5, 2.5, 3.5, 4.5])
+    array([ 0.,  2.,  2.,  4.,  4.])
+
+    So, for our application, we detect whether the above case occurs, and use
+    ``np.floor`` if so. It is sufficient to detect that the first coordinate
+    falls on 0.5 and that the second coordinate is 1.0 apart, since we assume
+    by construction that the inter-point distance is less than or equal to 1
+    and that all successive points are equidistant.
+
+    Parameters
+    ----------
+    coords : 1D array of float
+        The coordinates array. We assume that all successive values are
+        equidistant (``np.all(np.diff(coords) = coords[1] - coords[0])``)
+        and that this distance is no more than 1
+        (``np.abs(coords[1] - coords[0]) <= 1``).
+
+    Returns
+    -------
+    rounded : 1D array of int
+        The array correctly rounded for an indexing operation, such that no
+        successive indices will be more than 1 apart.
+
+    Examples
+    --------
+    >>> coords0 = np.array([0.5, 1.25, 2., 2.75, 3.5])
+    >>> _round_safe(coords0)
+    array([0, 1, 2, 3, 4])
+    >>> coords1 = np.arange(0.5, 8, 1)
+    >>> coords1
+    array([ 0.5,  1.5,  2.5,  3.5,  4.5,  5.5,  6.5,  7.5])
+    >>> _round_safe(coords1)
+    array([0, 1, 2, 3, 4, 5, 6, 7])
+    """
+    if coords[0] % 1 == 0.5 and coords[1] - coords[0] == 1:
```

Comment:
Why? Do you care about pretty rendering? leave that for a graphics library no?

Reply:
I care about evenly-spaced measurements.

Reply:
```
_round_safe(np.asarray(6))

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-17-94c5c8ecb32c> in <module>
----> 1 _round_safe(np.asarray(6))

<ipython-input-1-57179cafa7e7> in _round_safe(coords)
     35     array([0, 1, 2, 3, 4, 5, 6, 7])
     36     """
---> 37     if coords[0] % 1 == 0.5 and coords[1] - coords[0] == 1:
     38         _round_function = np.floor
     39     else:

IndexError: too many indices for array
```

Reply:
Well, that's actually not a thing that linspace can return, but it *can* return a length-0 1D array, which will also croak with an IndexError. I'll add a check for that. Thanks for the battle-testing! =)

Reply:
```
In [7]: np.linspace(2, 3, num=1)
Out[7]: array([2.])
```

Reply:
```
In [8]: np.linspace(6, 7, num=1)
Out[8]: array([6.])
```

Reply:
That's not the same as `np.array(6)`:

```
In [1]: np.array(6)                                                                
Out[1]: array(6)

In [2]: np.array(6).shape                                                          
Out[2]: ()

In [3]: np.linspace(6, 7, num=1).shape                                             
Out[3]: (1,)
```

But:

```

In [4]: np.linspace(6, 6, num=0).shape                                             
Out[4]: (0,)
```

which will cause an error in the current code.